qa/tasks/vstart_runner.py

```diff
@@ -388,10 +383,41 @@ def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
 
         return args
 
+    @staticmethod
+    def split_args(args):
+        # Since Python's shell simulation can only work when commands are
+        # provided as a list of argumensts...
+        if isinstance(args, str):
+            if '\n' in args:
+                args = args.split('\n')
+                args = [a.split() for a in args]
```

Comment:
BTW, will the following case happen ?

If the `args` is something like:  `"ls /mnt/cpehfs '\n' cd /mnt/cephfs/"`, after this it will be: `['ls', '/mnt/cephfs', 'cd', '/mnt/cephfs/']`.

Should we add a `;` between the two commands? like:
`['ls', '/mnt/cephfs', ';', 'cd', '/mnt/cephfs/']` or `['ls', '/mnt/cephfs', '&&', 'cd', '/mnt/cephfs/']`.

Reply:
I don't think that would work. Python's shell simulation would think of the latter elements as arguments for the first element.

Reply:
BTW, do you have the use case of it ?

Reply:
Just as I had expected, `;` is not interpreted as command separator -
```
2020-12-01T19:42:27.929 INFO:__main__:> ls ; ls
2020-12-01T19:42:27.942 INFO:__main__:> sudo umount /tmp/tmpkuz9qrta/mnt.0
2020-12-01T19:42:27.986 INFO:__main__:> rmdir -- /tmp/tmpkuz9qrta/mnt.0
2020-12-01T19:42:27.990 INFO:__main__:> ./bin/ceph config rm mon mon_allow_pool_delete
2020-12-01T19:42:28.368 INFO:__main__:> ./bin/ceph log Ended test tasks.cephfs.test_cephfs_shell.TestCD.test_cd_with_no_args
2020-12-01T19:42:29.497 INFO:__main__:Stopped test: test_cd_with_no_args (tasks.cephfs.test_cephfs_shell.TestCD)
Test that when cd is issued without any arguments, CWD is changed in 75.512632s
2020-12-01T19:42:29.497 INFO:__main__:test_cd_with_no_args (tasks.cephfs.test_cephfs_shell.TestCD)
2020-12-01T19:42:29.497 INFO:__main__:Test that when cd is issued without any arguments, CWD is changed ... ERROR
2020-12-01T19:42:29.497 INFO:__main__:
2020-12-01T19:42:29.498 INFO:__main__:======================================================================
2020-12-01T19:42:29.498 INFO:__main__:ERROR: test_cd_with_no_args (tasks.cephfs.test_cephfs_shell.TestCD)
2020-12-01T19:42:29.498 INFO:__main__:Test that when cd is issued without any arguments, CWD is changed
2020-12-01T19:42:29.498 INFO:__main__:
2020-12-01T19:42:29.498 INFO:__main__:----------------------------------------------------------------------
2020-12-01T19:42:29.498 INFO:__main__:Traceback (most recent call last):
2020-12-01T19:42:29.498 INFO:__main__:  File "/home/rishabh/repos/ceph/minor/qa/tasks/cephfs/test_cephfs_shell.py", line 495, in test_cd_with_no_args
2020-12-01T19:42:29.498 INFO:__main__:    self.mount_a.client_remote.run(args='ls\nls')
2020-12-01T19:42:29.498 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 423, in run
2020-12-01T19:42:29.498 INFO:__main__:    return self._do_run(**kwargs)
2020-12-01T19:42:29.498 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 492, in _do_run
2020-12-01T19:42:29.498 INFO:__main__:    proc.wait()
2020-12-01T19:42:29.498 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 208, in wait
2020-12-01T19:42:29.498 INFO:__main__:    raise CommandFailedError(self.args, self.exitstatus)
2020-12-01T19:42:29.498 INFO:__main__:teuthology.exceptions.CommandFailedError: Command failed with status 2: ['ls', ';', 'ls']
```

Return value 2 stands `No such file or directory`.

Reply:
> BTW, do you have the use case of it ?

Use case for `;` is not interpreted as separator or something else?

Reply:
@lxbsz 
Ah, I see misunderstanding  here...

> If the args is something like: "ls /mnt/cpehfs '\n' cd /mnt/cephfs/", after this it will be: ['ls', '/mnt/cephfs', 'cd', '/mnt/cephfs/'].


Command like "ls /mnt/cpehfs '\n' cd /mnt/cephfs/" would be split into [['ls', '/mnt/cephfs'], ['cd', '/mnt/cephfs/']].


```
if '\n' in args: # if args = 'ls /mnt/cpehfs '\n' cd /mnt/cephfs/'
                args = args.split('\n') # args would become ['ls /mnt/cpehfs', 'cd /mnt/cephfs/']
                args = [a.split() for a in args] # and now it would become [['ls', '/mnt/cephfs'], ['cd', '/mnt/cephfs/']]
```

Reply:
Okay, that looks good.